### PR TITLE
Handle Shard Gemshop in Sell-loot

### PR DESCRIPTION
**Pull request by jandersson**
The shard gem shop changes the person appraising the gems depending on time of day. At night it is Veilex, in the morning it is Fatima. In base-town we have Fatima listed, so if we see Veilex in the shop we temporarily overwrite Fatima.

### CODE REVIEW

**Comment by rpherbig** [comment]
I'm wondering if a better approach would be to change base-town... maybe a list of gemshop NPCs? What do you think?

**Comment by jandersson** [comment]
That sounds like a better approach as there might be other towns with similar situations. However that line of code in the PR would remain similar though, right? We check against the list instead of against the string

**Comment by jandersson** [comment]
I noticed I deleted some spaces in the empty fields. Are those considered good style or does that change the datatype?

**Comment by rcuhljr** [comment]
They'll just be ignored, I have 'remove trailing whitespace on save' set in my editor so they wouldn't last long anyways.

File: sell-loot.lic [hunk 1]
```diff
@@ -76,11 +76,18 @@ class SellLoot
...
-    fput("ask #{@hometown['gemshop']['name']} for #{adj} pouch")
+    clerk = which_clerk(@hometown['gemshop']['name'])
+    fput("ask #{clerk} for #{adj} pouch")
     fput("put my pouch in my #{container}")
   end
 
+  def which_clerk(clerks)
+    return clerks if clerks.is_a? String
+
+    clerks.select! { |clerk| DRRoom.npcs.include?(clerk) }
```
Comment (rpherbig) [inline sell-loot.lic]: These two lines can be replaced with `clerks.find { |clerk| DRRoom.npcs.include?(clerk) }`

Comment (jandersson) [inline sell-loot.lic]: Perfect, thank you

Comment (jandersson) [inline sell-loot.lic]: Would this require turning every gemshop entry into a list? I was trying to avoid that with the return if is a String statement

Comment (jandersson) [inline sell-loot.lic]: Or do you mean the two lines to be the select! and clerks.first

Comment (rcuhljr) [inline sell-loot.lic]: I'd move towards making it a list instead of relying on type checking, but these aren't really user editable yaml files so it's low concern that they might get confused. However the whole method could just be `clerks.is_a?(String) ? clerks : clerks.find {...`

Comment (jandersson) [inline sell-loot.lic]: OK I will test out the changes. Thanks for the advice!

Comment (jandersson) [inline sell-loot.lic]: So, just to put my thoughts out there... Since this is semistructured data, we dont have to make every attribute a list because of one outlier. We have to adapt the program to handle this outlier in either case. However the data entry becomes much easier if its just a field for a string, which i think all the other gemshops are.

File: profiles/base-town.yaml [hunk 2]
```diff
@@ -16,16 +16,25 @@ Crossing:
     name: appraiser
   tannery:
     id: 8266
-    name: falken
+    name: Falken
   npc_empath:
     id: 6218
+    action: lie
```
Comment (rpherbig) [inline profiles/base-town.yaml]: Out of curiosity, why did you add an action to `npc_empath`?
